crates/solver/src/settlement.rs

```diff
@@ -299,6 +318,77 @@ impl Settlement {
         }
     }
 
+    // Checks whether the settlement prices do not deviate more than max_settlement_price_deviation from the auction prices on certain pairs.
+    pub fn satisfies_price_checks(
+        &self,
+        auction_id: u64,
+        solver_name: &str,
+        external_prices: &ExternalPrices,
+        max_settlement_price_deviation: &Ratio<BigInt>,
+        tokens_to_satisfy_price_test: &TokensForPriceCheck,
+    ) -> bool {
+        if matches!(tokens_to_satisfy_price_test, TokensForPriceCheck::Tokens(token_list) if token_list.is_empty())
+        {
+            return true;
+        }
+        // The following check is quadratic in run-time, although a similar check with linear run-time would also be possible.
+        // For the linear implementation, one would have to find a unique scaling factor that scales the external prices into
+        // the settlement prices. Though, this scaling factor is not easy to define, if reference tokens like ETH are missing.
+        // Since the checks would heavily depend on this scaling factor, and its
+        // derivation is non-trivial, we decided to go for the implementation with quadratic run time. Settlements
+        // will not have enough tokens, such that the run-time is important.
+        self
+            .clearing_prices()
+            .iter()
+            .combinations(2)
+            .all(|clearing_price_vector_combination| {
+                let (sell_token, sell_price) = clearing_price_vector_combination[0];
+                let clearing_price_sell_token = sell_price.to_big_rational();
+                let (buy_token, buy_price) = clearing_price_vector_combination[1];
+                let clearing_price_buy_token = buy_price.to_big_rational();
+
+                if matches!(tokens_to_satisfy_price_test, TokensForPriceCheck::Tokens(token_list) if (!token_list.contains(sell_token)) || !token_list.contains(buy_token))
```

Comment:
Same here. I find this match confusing because of the content on the right. I think if you make a struct as described above, it could implement some method that moves the awkward logic of these match statements into one place.